mlflow/utils/process.py

```diff
@@ -3,57 +3,59 @@
 
 
 class ShellCommandException(Exception):
-    pass
+    @classmethod
+    def from_completed_process(cls, process):
+        lines = [
+            f"Non-zero exit code: {process.returncode}",
+            f"Command: {process.args}",
+        ]
+        if process.stdout:
+            lines += [
+                "",
+                "STDOUT:",
+                process.stdout,
+            ]
+        if process.stderr:
+            lines += [
+                "",
+                "STDERR:",
+                process.stderr,
+            ]
+        return cls("\n".join(lines))
 
 
-def exec_cmd(
-    cmd, throw_on_error=True, env=None, capture_output=True, cwd=None, cmd_stdin=None, **kwargs
+def _exec_cmd(
+    cmd,
+    *,
+    throw_on_error=True,
+    extra_env=None,
+    capture_output=True,
+    **kwargs,
 ):
     """
-    Runs a command as a child process.
+    A convenience wrapper of `subprocess.run` for running a command from a Python script.
 
-    A convenience wrapper for running a command from a Python script.
-    Keyword arguments:
-    cmd -- the command to run, as a list of strings
-    throw_on_error -- if true, raises an Exception if the exit code of the program is nonzero
-    env -- additional environment variables to be defined when running the child process
-    cwd -- working directory for child process
-    capture_output -- If True, stdout and stderr will be captured and included in an exception
-                      message on failure; if False, these streams won't be captured.
-    cmd_stdin -- if specified, passes the specified string as stdin to the child process.
-
-    Note on the return value: If stream_output is true, then only the exit code is returned. If
-    stream_output is false, then a tuple of the exit code, standard output and standard error is
-    returned.
+    :param cmd: The command to run, as a list of strings.
+    :param throw_on_error: If True, raises an Exception if the exit code of the program is nonzero.
+    :param extra_env: Extra environment variables to be defined when running the child process.
```

Comment:
Good renaming!

Reply:
Shall we forbid user to set "env" argument ?
if both `extra_env` and `env` argument existing, it is ambiguous

Reply:
I think we should keep both because `env` is useful when you want to exclude some environmental variables.